tokio/src/sync/broadcast.rs

```diff
@@ -247,17 +246,14 @@ pub enum TryRecvError {
 /// Data shared between senders and receivers
 struct Shared<T> {
     /// slots in the channel
-    buffer: Box<[Slot<T>]>,
+    buffer: Box<[RwLock<Slot<T>>]>,
```

Comment:
A lock per item in the queue? That sounds excessive, unless the channel is heavy contended.

Reply:
locks are cheap w/ parking lot (atomic usize). This is how it was structured before :shrug: